### PR TITLE
Cli flag usage refactor

### PR DESCRIPTION
**Pull request by tcar121293**
## Description
Refactor CLI flag usage

## Related Issue Or Context

Closes: #126 

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation

## Checklist:

- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have ensured that all acceptance criteria (or expected behavior) from issue are met
- [ ] I have updated the documentation locally and in chainbridge-docs.
- [ ] I have added tests to cover my changes.
- [ ] I have ensured that all the checks are passing and green, I've signed the CLA bot

### CODE REVIEW

File: chains/evm/cli/admin/add-relayer.go [hunk 1]
```diff
@@ -22,59 +21,71 @@ var addRelayerCmd = &cobra.Command{
...
-	log.Info().Msgf("Setting address %s as relayer on bridge %s", relayer.String(), bridge.String())
-	addRelayerInput, err := calls.PrepareAddRelayerInput(relayer)
+	log.Info().Msgf("Setting address %s as relayer on bridge %s", relayerAddr.String(), bridgeAddr.String())
+	addRelayerInput, err := calls.PrepareAddRelayerInput(relayerAddr)
 	if err != nil {
 		log.Error().Err(err)
 		return err
 	}
 
 	_, err = calls.Transact(ethClient, txFabric, &bridge, addRelayerInput, gasLimit, big.NewInt(0))
```
Comment (MakMuftic) [inline chains/evm/cli/admin/add-relayer.go:86]: This `&bridge` reference should be `&bridgeAddr`

File: chains/evm/cli/flags/utils.go [hunk 2]
```diff
@@ -58,3 +60,23 @@ func defineSender(cmd *cobra.Command) (*secp256k1.Keypair, error) {
...
+		resourceID = resourceID[2:]
+	}
+	resourceIdBytes, err := hex.DecodeString(resourceID)
+	if err != nil {
+		return [32]byte{}, err
+	}
+	return calls.SliceTo32Bytes(resourceIdBytes), nil
+}
+
+func CheckRequiredFlags(cmd *cobra.Command, flags ...string) {
```
Comment (MakMuftic) [inline chains/evm/cli/flags/utils.go:75]: I think proper naming would be something like `MarkAllFlagsAsRequired` because from original name I would assume that this functions checks if flags are required.

File: chains/evm/cli/admin/set-deposit-nonce.go [hunk 3]
```diff
@@ -24,49 +23,50 @@ This nonce cannot be less than what is currently stored in the contract`,
...
-	BindSetDepositNonceFlags(setDepositNonceCmd)
+	BindSetDepositNonceFlags()
 }
 
-func SetDepositNonceEVMCMD(cmd *cobra.Command, args []string, txFabric calls.TxFabric) error {
-	domainId, err := cmd.Flags().GetUint8("domainId")
-	if err != nil {
-		return err
+func validateSetDepositNonceFlags(cmd *cobra.Command, args []string) error {
+	if !common.IsHexAddress(Relayer) {
```
Comment (MakMuftic) [inline chains/evm/cli/admin/set-deposit-nonce.go:49]: **Relayer** flag is not used in `set-deposit-nonce`

Comment (MakMuftic) [reply chains/evm/cli/admin/set-deposit-nonce.go:49]: Please make sure to double check this (unused flags) for everything.

File: chains/evm/cli/flags/utils.go [hunk 4]
```diff
@@ -58,3 +60,23 @@ func defineSender(cmd *cobra.Command) (*secp256k1.Keypair, error) {
 	var AliceKp = keystore.TestKeyRing.EthereumKeys[keystore.AliceKey]
 	return AliceKp, nil
 }
+
+func ProcessResourceID(resourceID string) ([32]byte, error) {
```
Comment (MakMuftic) [inline chains/evm/cli/flags/utils.go:64]: I think this should return our new type for resource ID

File: chains/evm/cli/bridge/register-resource.go [hunk 5]
```diff
@@ -22,62 +21,74 @@ var registerResourceCmd = &cobra.Command{
...
 		return RegisterResourceCmd(cmd, args, txFabric)
 	},
+	Args: func(cmd *cobra.Command, args []string) error {
+		err := validateRegisterResourceFlags(cmd, args)
+		if err != nil {
+			return err
+		}
+
+		err = processRegisterResourceFlags(cmd, args)
+		if err != nil {
```
Comment (MakMuftic) [inline chains/evm/cli/bridge/register-resource.go:31]: There is no need for this check, you can just return `err` as it will be `nil` if everything is ok

File: chains/evm/cli/bridge/register-resource.go [hunk 6]
```diff
@@ -22,62 +21,74 @@ var registerResourceCmd = &cobra.Command{
...
+}
+
+func processRegisterResourceFlags(cmd *cobra.Command, args []string) error {
+	var err error
+	handlerAddr = common.HexToAddress(Handler)
+	targetContractAddr = common.HexToAddress(Target)
+	bridgeAddr = common.HexToAddress(Bridge)
+
+	resourceIdBytesArr, err = flags.ProcessResourceID(ResourceID)
+	if err != nil {
```
Comment (MakMuftic) [inline chains/evm/cli/bridge/register-resource.go:70]: There is no need for this check, you can just return `err` as it will be `nil` if everything is ok

File: chains/evm/cli/bridge/set-burn.go [hunk 7]
```diff
@@ -22,44 +21,55 @@ var setBurnCmd = &cobra.Command{
...
+	if !common.IsHexAddress(TokenContract) {
+		return fmt.Errorf("invalid token contract address %s", TokenContract)
+	}
+	if !common.IsHexAddress(Bridge) {
+		return fmt.Errorf("invalid bridge address %s", Bridge)
+	}
+	return nil
+}
 
+var tokenContractAddr common.Address
```
Comment (MakMuftic) [inline chains/evm/cli/bridge/set-burn.go:59]: Why is this here ?

File: chains/evm/cli/erc20/add-minter.go [hunk 8]
```diff
@@ -22,47 +22,57 @@ var addMinterCmd = &cobra.Command{
...
+	if !common.IsHexAddress(Erc20Address) {
+		return errors.New("invalid erc20Address address")
+	}
+	if !common.IsHexAddress(Handler) {
+		return errors.New("invalid minter address")
+	}
+	return nil
+}
+
+var handlerAddr common.Address
```
Comment (MakMuftic) [inline chains/evm/cli/erc20/add-minter.go:55]: Same, why is this here ?

File: README.md [hunk 9]
```diff
@@ -423,7 +423,7 @@ Usage:
 Flags:
       --erc20Address string   ERC20 contract address
   -h, --help                  help for add-minter
-      --minter string         address of minter
+      --handler string        handler contract address
```
Comment (P1sar) [inline README.md:426]: Arn't we decide to leave it as minter, but rather update the Readme, so users understand that in case of bridging role of minter should be given to the Handler?

File: chains/evm/cli/erc20/approve.go [hunk 10]
```diff
@@ -23,69 +22,77 @@ var approveCmd = &cobra.Command{
...
 		return ApproveCmd(cmd, args, txFabric)
 	},
+	Args: func(cmd *cobra.Command, args []string) error {
+		err := validateApproveFlags(cmd, args)
+		if err != nil {
+			return err
+		}
+
+		err = processApproveFlags(cmd, args)
+		if err != nil {
```
Comment (MakMuftic) [inline chains/evm/cli/erc20/approve.go:32]: You can just return `error`

**Comment by MakMuftic** [comment]
@tcar121293 Please make sure that you include util CLI commands in this refactor (`chains/evm/cli/utils`)

**Review by P1sar — CHANGES_REQUESTED** [review]

File: chains/evm/cli/erc20/add-minter.go (hunk 8 above)
Comment (tcar121293) [reply chains/evm/cli/erc20/add-minter.go:55]: The same as `tokenContractAddr`
